scex-core/src/test/scala/com/avsystem/scex/compiler/JavaScexCompilerTest.scala

```diff
@@ -26,6 +33,72 @@ class JavaScexCompilerTest extends AnyFunSuite with CompilationTest {
       .profile(createProfile(acl))
       .get
 
-    assert(180.0 == cexpr(context))
+    // when expression is evaluated
+    val result = cexpr(context)
+
+    // then the result is correct
+    assert(math.Pi.toDegrees == result)
   }
+
+  TypedVariables.IllegalCharsInVarName.foreach { char =>
+    test(s"typed variables fail if char [code=${char.toInt}, escaped=${escapedChar(char)}] is in var name test") {
+      // given a context with a typed variable with illegal char in name
+      val acl = PredefinedAccessSpecs.basicOperations
+      val variableName = s"JP2NadajePoRazPierwszy$char"
+      val context = SimpleContext(())
+      context.setTypedVariable(variableName, math.Pi)
+
+      // given expression that uses the typed variable
+      val expr = s"#`$variableName`.toDegrees"
+
+      // then compilation should fail
+      withClue(s"[code=${char.toInt}] character in variable name [$variableName] should not be allowed") {
+        assertThrows[CompilationFailedException] {
+          compiler.buildExpression
+            .contextType(ScalaTypeTokens.create[SimpleContext[Unit]])
+            .resultType(classOf[Double])
+            .expression(expr)
+            .template(false)
+            .variableClass(variableName, classOf[Double])
+            .profile(createProfile(acl))
+            .get
```

Comment:
Very similar to SuspiciousCharsAllowedInVariableNames testcase, consider extracting a helper method to deduplicate